src/jit/morph.cpp

```diff
@@ -10998,6 +10998,7 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                 //
                 // a % b = a - (a / b) * b;
                 //
+                assert(!optValnumCSE_phase);
```

Comment:
Presumably this is only an assert because global morph takes care of all `rem` nodes and this transform never runs again, the JIT doesn't create `rem` nodes by itself.